### ISSUE TITLE
Craftweaker Support

### ISSUE THREAD

**Issue by bookerthegeek** [BuildCraft/BuildCraftCompat#81]
## Heyo ##
Just a small reminder (in the extremely unlikely event you forgot, ha!) that minetweaker support has not been ported up yet.

**Comment by loordgek** [comment]
no BC is using json now

**Comment by AlexIIL** [comment]
Crafttweaker is much simpler to configure than multiple json files, so it will be ported in the future.

**Comment by bookerthegeek** [comment]
@AlexIIL Thank you.


@loordgek ....
1. Internal JSON recipes != CraftTweaker Functionality
2. Not Editable by User (yet)
3. JSON = Complicated, ZenScript = Easy Learning Curve
4 ZenScript is easily editable in [NotePad++](https://notepad-plus-plus.org/) w/[syntax highlighting](https://minecraft.curseforge.com/projects/zenscript-syntax-highlighter-for-notepad), while Json require an additional editor. If you know of a free easy to use syntax highlighting program for Json that also supports from multiple mods please let me know as I would love to know of it.
4. Not as powerful as CraftTweaker. I'd love for you to prove me wrong on this one. If you could explain how to do a for/if loop iterating  over an array of arrays in Json I will be very impressed.

**Comment by AlexIIL** [comment]
Perhaps my previous comment was a bit strong.

Crafttweaker support will be added because it's an alternative that some people would probably prefer over JSON. It isn't high on our priority lists, and probably won't happen until late beta.

````
Not Editable by User (yet)
````
This support won't be added before buildcraft's own recipe loading system is capable of loading from per-pack and per-world, so this point is fairly moot in regards to buildcraft compat adding crafttweaker support.
````
JSON = Complicated, ZenScript = Easy Learning Curve
````
Every JSON recipe file for a particular recipe type (pretty much) follows the same format - if you copy an existing recipe file and modify it then you can see exactly what you have and what you need to do - the string keys are just given to you directly. By contrast Zen script is a programming language where conditionals and loops are *by design* more complex than just the raw data.
````
ZenScript is easily editable in NotePad++ w/syntax highlighting, while JSON require an additional editor
````
As a simpler format, JSON just doesn't need that kind of support. (Although as JSON is such a widely used format for other things JSON syntax highlighting tools do exist). Using a basic text editor is enough to configure a json recipe file. In order to know the order of arguments and which function to call you need a wiki.
````
Not as powerful as CraftTweaker. I'd love for you to prove me wrong on this one. If you could explain how to do a for/if loop iterating over an array of arrays in Json I will be very impressed.
````
*Technically* arrays are only one way of solving a problem, rather than being the *only* solution. If you need to iterate you just need to make a lot of JSON files. (Yes it's annoying and obviously not ideal, but it's not impossible to have the same results in non-dynamic cases).
Although yes, this sort of functionality is pretty much the main hard benefit of using crafttweaker over JSON.

**Comment by bookerthegeek** [comment]
AlexIIL Thank you for that explanation. Re-reading my previous commit I see it could come across a bit differently then I intended. Sorry about that loordgek. Can I ask for a recommendation for a good desktop JSON editor though as I have a feeling some mods will prefer that to craftweaker.

I look forward to when it is added, and will patiently await that time.

And yes, I much prefer a simple loop statement then manually righting out every recipe. For example [THIS](https://gist.github.com/bookerthegeek/2c2175cd5a618ae1b1f2285a1f6c19f0)

**Comment by AlexIIL** [comment]
Yeah loops are amazing and honestly I want to make some way of doing this via JSON or something for normal crafting recipes. Although I don't want craft tweaker to be required for tweaking any buildcraft recipe.

**Comment by AlexIIL** [comment]
Er, I'm not sure of a good JSON editor myself as I always just use the eclipse plugin instead :/
